style: added conditional text to /available/; Closes: #817

9921622 opened the pull request:
I didn't do a test for the text that pops up for hidden quests because I still have no idea what constitutes as an "available" quest.

However, I made sure to manually test everything, and it seems everything works as intended

tylerecouture commented on src/quest_manager/templates/quest_manager/quests.html:
```diff
@@ -71,6 +71,14 @@
+          <p>You have no new quests available, but you do have hidden quests which you can view by hitting the 'Show Hidden Quests' button above.</p>
+        {% elif awaiting_submission %}
+          <p>New quests will become available after your teacher approves your submission!</p>
+        {% endif %}
```
There should be a final {% else %} here with: "There are currently no new quest available to you!"

tylerecouture commented on src/quest_manager/tests/test_views.py:
```diff
@@ -159,6 +161,97 @@ def test_start(self):
+        Quest.objects.all().delete()
+        
+        # # #quest should be in 'hidden' ===
+        # quest = Quest.objects.create(
```
Remove all this commented out code, or put a big comment on top explaining why it's here and what it's waiting for before it can be used.

tylerecouture replied:
Actually, maybe just delete this code and add a new issue on github for "Add tests to 'test_student_no_quests_help_text' " and describe the needed missing tests in the issue

9921622 replied:
submitted an issue refrencing this: https://github.com/bytedeck/bytedeck/issues/1011

tylerecouture commented on src/quest_manager/tests/test_views.py:
```diff
@@ -159,6 +161,97 @@ def test_start(self):
         # the view should have redirect to the same submission:
         self.assertRedirects(response, sub.get_absolute_url())
 
+    @tag("do")
```
haha, remove the tag and import!

9921622 replied:
removed

tylerecouture commented on src/quest_manager/tests/test_views.py:
```diff
@@ -159,6 +161,97 @@ def test_start(self):
+        self.assertTrue(success)
+
+        # condition to see any of the 3
+        self.assertTrue(user.profile.has_current_course and not user.is_staff)
```
Excellent comment.  I know it seems small and silly, but allowed me to instantly understand why you were asserting this.

tylerecouture commented on src/quest_manager/tests/test_views.py:
```diff
@@ -159,6 +161,97 @@ def test_start(self):
+        self.assertTrue(user.profile.has_current_course and not user.is_staff)
+
+        # make sure no available quests
+        Quest.objects.all().delete()
```
lol, harsh, but fair.

tylerecouture commented on src/quest_manager/tests/test_views.py:
```diff
@@ -159,6 +161,97 @@ def test_start(self):
+
+        response = self.client.get(url)
+        self.assertContains(response, "No quests are available.")
+        flake8string = "You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above."
```
Why are you calling this variable `flake8string`?  what does this have to do with flake8? should be `expected_text` or something more informative.

9921622 replied:
renamed to 'help_text'

tylerecouture commented on src/quest_manager/views.py:
```diff
@@ -372,8 +372,10 @@ def quest_list(request, quest_id=None, template="quest_manager/quests.html"):
     # num_inprogress = QuestSubmission.objects.all_not_completed(request.user).count()
     # num_completed = QuestSubmission.objects.all_completed(request.user).count()
 
+    awaiting_submission = QuestSubmission.objects.filter(is_approved=False, is_completed=True).exists()
```
What does "awaiting submission" mean?  Did you mean "awaiting approval"?  Also, this is not filtered for the current user, so it will be getting ALL submissions from every user. (I guess this one really needed a test =D

9921622 replied:
changed name and now added user=request.user to the filter, as well as a test

tylerecouture reviewed (changes requested):
Great work here, a few nuances and tweaks requested.  Also currently the "awaiting_submission" won't work as intended I suspect

9921622 commented:
Added:
- fixed awaiting_submission by passing an extra argument for user in the filter method
- changed awaiting_submission  to awaiting_approval
- added a test that tests awaiting_submission  if it has submissions from different users
- added an extra help text if user completed all quests with nothing left + tests
-
